src/sentry/plugins/providers/integration_repository.py

```diff
@@ -27,14 +27,43 @@ def __init__(self, id):
 
     def dispatch(self, request, organization, **kwargs):
         try:
+            # get data via API (so we know it's up to date)
             config = self.get_repository_data(organization, request.DATA)
+
+        except Exception as e:
+            return self.handle_api_error(e)
+
+        else:
+            # check to see if we already have this repository, and update any
+            # data if applicable
+            repo = Repository.objects.filter(
+                organization_id=organization.id,
+                provider=self.id,
+                external_id=config.get('external_id')
+            ).first()
+            if repo is not None:
+                updated = self.update_repository_data(repo, config)
+                if not updated:
+                    return Response({
+                        'errors': {
+                            '__all__': 'That repository already exists'
```

Comment:
This (attempting to add a repo which hasn't changed at all) probably won't ever happen, given https://github.com/getsentry/sentry/pull/9288, but can't hurt to handle it.